src/pymonzo/transactions/schemas.py

```diff
@@ -175,6 +211,12 @@ def empty_str_to_none(cls, v: str) -> Optional[str]:
         """Convert empty strings to `None`."""
         return empty_str_to_none(v)
 
+    @field_validator("counterparty", mode="before")
+    @classmethod
+    def empty_dict_to_none(cls, v: dict) -> Optional[dict]:
+        """Convert empty dict to `None`."""
+        return empty_dict_to_none(v)
```

Comment:
So do some transactions have `counterparty` set to `None`, some to an empty dict `{}` and some to the schema specified in `MonzoTransactionCounterparty`?

Reply:
Either the empty dict or the schema specified in MonzoTransactionCounterparty... 

I've not used pydantic before so maybe there are better ways of achieving what I've done...

Reply:
I think I've cleaned this up now...

Reply:
Well, that's not great on Monzo's side... OK, let me have a quick look into this and get back to you.